.github/workflows/ci.yml

```diff
@@ -23,4 +23,4 @@ jobs:
       - run: cargo build --verbose
       - run: cargo test --verbose
       - run: cargo clippy
-      - run: cargo fmt --check
+      - run: cargo fmt --check -- --config error_on_unformatted=true,error_on_line_overflow=true
```

Comment:
This is a nice addition – thanks!

Given some of the weird things that happen when it tries to break up long comments (like the big URLs below), I wonder if we want `error_on_unformatted` to be true or not?

I generally agree that long lines are undesirable, but having it reformat comments and strings can be a little odd. Can we just have those be warnings?

Reply:
I believe there is a separate limit we can set for comments so we could look into that, but `error_on_unformatted` affects more like spaces left behind on lines that rustfmt for some reason doesn't know how to deal with, so that should probably stay enabled. Cargo fmt doesn't reformat strings and comments by default ( I believe there is some option to enable that tho at least in preview ), all the changes that I applied here were made manually cuz rustfmt can't do them itself, that's why they were never done before as well I believe

Reply:
Oh error on line overflow also ignores comments & string literals so that should be fine (at least according to the docs )

Reply:
Yea having an option to set it to error on trailing whitespace & warn on everything else would probably be nice

Reply:
we might just want 

```bash
cargo fmt --check -- --config=error_on_line_overflow=true,wrap_comments=true,comment_width=120
```

and then configure the comment width to something higher. This should also ignore any comments containing urls.